lively.user/user-flap.cp.js

```diff
@@ -190,16 +217,21 @@ class UserFlapModel extends ViewModel {
   }
 
   showGuestUser () {
-    const { leftUserLabel, rightUserLabel, avatar } = this.ui;
-    connect(leftUserLabel, 'onMouseDown', this, 'login');
-    leftUserLabel.tooltip = 'Login with GitHub';
-    rightUserLabel.tooltip = '';
-    disconnect(rightUserLabel, 'onMouseDown', this, 'logout');
-    leftUserLabel.nativeCursor = 'pointer';
-    rightUserLabel.nativeCursor = 'auto';
-    leftUserLabel.textAndAttributes = Icon.textAttribute('github');
-    rightUserLabel.textString = 'guest';
-    avatar.loadUrl('https://s.gravatar.com/avatar/d41d8cd98f00b204e9800998ecf8427e?s=160', false);
+    const { loginButton, leftUserLabel, rightUserLabel, avatar } = this.ui;
+    if (this.withLoginButton) {
+      leftUserLabel.visible = rightUserLabel.visible = avatar.visible = false;
+      loginButton.visible = loginButton.isLayoutable = true;
+    } else {
+      connect(leftUserLabel, 'onMouseDown', this, 'login');
+      leftUserLabel.tooltip = 'Login with GitHub';
+      rightUserLabel.tooltip = '';
+      disconnect(rightUserLabel, 'onMouseDown', this, 'logout');
```

Comment:
Instead of swapping the connections manually, it's better to just to the dispatch inside of a method like `toggleLoggedInUser()` or something.